v2/dtos/deviceprofile.go

```diff
@@ -54,3 +57,76 @@ func FromDeviceProfileModelToDTO(deviceProfile models.DeviceProfile) DeviceProfi
 		CoreCommands:    FromCommandModelsToDTOs(deviceProfile.CoreCommands),
 	}
 }
+
+func ValidateDeviceProfileDTO(profile DeviceProfile) error {
+	// deviceResources should not duplicated
+	for i := 0; i < len(profile.DeviceResources); i++ {
+		for j := i + 1; j < len(profile.DeviceResources); j++ {
+			if profile.DeviceResources[i].Name == profile.DeviceResources[j].Name {
+				return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("device resource %s is duplicated", profile.DeviceResources[j].Name), nil)
+			}
+		}
+	}
```

Comment:
These are old style for loop blocks. 
Please redo these blocks with Go style range and map which more concise and easier to follow and verify code.

```
dupCheck := make(map[string] bool)
for _, resource:= range profile.DeviceResources {
   if dupCheck[resource.name] {
       return error....
   }
   dupCheck[resource.name] = true
}

// reset map for next check
dupCheck = make(map[string] bool)
```

Reply:
Modified.